modules/phpfpm/phpfpm.go

```diff
@@ -0,0 +1,139 @@
+package phpfpm
+
+import (
+	"math"
+	"time"
+
+	"github.com/netdata/go.d.plugin/pkg/stm"
+	"github.com/netdata/go.d.plugin/pkg/web"
+
+	"github.com/netdata/go-orchestrator/module"
+)
+
+func init() {
+	creator := module.Creator{
+		Defaults: module.Defaults{
+			Disabled: true,
+		},
+		Create: func() module.Module { return New() },
+	}
+
+	module.Register("phpfpm", creator)
+}
+
+const (
+	defaultURL         = "http://127.0.0.1/status?full&json"
+	defaultHTTPTimeout = time.Second
+)
+
+// Config is the php-fpm module configuration.
+type Config struct {
+	web.HTTP `yaml:",inline"`
+}
+
+// Phpfpm collets php-fpm metrics.
+type Phpfpm struct {
+	module.Base
+
+	Config `yaml:",inline"`
+
+	client *client
+}
+
+// New returns a php-fpm module with default values.
+func New() *Phpfpm {
+	config := Config{
+		HTTP: web.HTTP{
+			Request: web.Request{UserURL: defaultURL},
+			Client:  web.Client{Timeout: web.Duration{Duration: defaultHTTPTimeout}},
+		},
+	}
+
+	return &Phpfpm{
+		Config: config,
+	}
+}
+
+// Init makes initialization.
+func (n *Phpfpm) Init() bool {
+	if err := n.ParseUserURL(); err != nil {
+		n.Errorf("error on parsing url '%s' : %v", n.UserURL, err)
+		return false
+	}
+
+	if n.URL.Host == "" {
+		n.Error("URL is not set")
+		return false
+	}
+
+	client, err := web.NewHTTPClient(n.Client)
+	if err != nil {
+		n.Error(err)
+		return false
+	}
+
+	n.client = newClient(client, n.Request, )
+
+	n.Debugf("using URL %s", n.URL)
+	n.Debugf("using timeout: %s", n.Timeout.Duration)
+
+	return true
+}
+
+// Check checks the module can collect metrics.
+func (n *Phpfpm) Check() bool {
+	return len(n.Collect()) > 0
+}
+
+// Charts creates Charts.
+func (*Phpfpm) Charts() *Charts {
+	return charts.Copy()
+}
+
+// Collect returns collected metrics.
+func (n *Phpfpm) Collect() map[string]int64 {
+	status, err := n.client.Status()
+	if err != nil {
+		n.Error(err)
+		return nil
+	}
+
+	data := stm.ToMap(status)
+	if len(status.Processes) == 0 {
+		return data
+	}
+
+	statProcesses(data, status.Processes, "ReqDur", func(p proc) int64 { return p.Duration })
+	statProcesses(data, status.Processes, "ReqCpu", func(p proc) int64 { return int64(p.CPU) })
+	statProcesses(data, status.Processes, "ReqMem", func(p proc) int64 { return p.Memory })
```

Comment:
please move collect logic to the `collect.go`

https://github.com/netdata/go.d.plugin/blob/e2059a14dc50d2c4acd86e64dfde4b5f1f99a15c/modules/vsphere/vsphere.go#L187-L199

Reply:
Is there any particular reason for this? From a Go perspective, this is not needed or probably wanted. You are merely splitting code a) into a function and b) into a file when the logic is in fact really simple.

Reply:
The reason is having `collect.go` as an entry point for collecting logic in all modules